tests/roles/manila_adoption/tasks/main.yaml

```diff
@@ -1,55 +1,17 @@
-- name: deploy podified Manila with cephfs backend
-  when: manila_backend == "cephfs"
-  ansible.builtin.shell: |
-    {{ shell_header }}
-    {{ oc_header }}
-    oc patch openstackcontrolplane openstack --type=merge --patch '
-    spec:
-      manila:
-        enabled: true
-        apiOverride:
-          route: {}
-        template:
-          databaseInstance: openstack
-          databaseAccount: manila
-          manilaAPI:
-            customServiceConfig: |
-              [DEFAULT]
-              enabled_share_protocols=cephfs
-            replicas: 1
-            networkAttachments:
-            - internalapi
-            override:
-              service:
-                internal:
-                  metadata:
-                    annotations:
-                      metallb.universe.tf/address-pool: internalapi
-                      metallb.universe.tf/allow-shared-ip: internalapi
-                      metallb.universe.tf/loadBalancerIPs: 172.17.0.80
-                  spec:
-                    type: LoadBalancer
-          manilaScheduler:
-            replicas: 1
-          manilaShares:
-            share1:
-              customServiceConfig: |
-                [DEFAULT]
-                enabled_share_backends=cephfs
-                enabled_share_protocols=cephfs
-                [cephfs]
-                driver_handles_share_servers=False
-                share_backend_name=cephfs
-                share_driver=manila.share.drivers.cephfs.driver.CephFSDriver
-                cephfs_conf_path=/etc/ceph/ceph.conf
-                cephfs_auth_id=openstack
-                cephfs_cluster_name=ceph
-                cephfs_volume_mode=0755
-                cephfs_protocol_helper_type=CEPHFS
-              replicas: 1
-              networkAttachments:
-              - storage
-    '
+- name: Deploy Podified Manila
+  when: manila_backend == "cephfs" or manila_backend == "cephnfs"
```

Comment:
Both `native_cephfs` and `cephnfs` use the same template, that is rendered with the appropriate config and applied through `oc patch ..`